lib/common/crmcommon_private.h

```diff
@@ -300,6 +300,7 @@ typedef struct pcmk__ipc_methods_s {
 struct pcmk_ipc_api_s {
     enum pcmk_ipc_server server;          // Daemon this IPC API instance is for
     enum pcmk_ipc_dispatch dispatch_type; // How replies should be dispatched
+    // @COMPAT Deprecated and unused since 3.0.1
     size_t ipc_size_max;                  // maximum IPC buffer size
```

Comment:
I think you're okay to just drop this. `struct pcmk_ipc_api_s` is library-private. It gets `typedef`'d to `pcmk_ipc_api_t`, but the struct definition itself is opaque.

> Alternatively, the structure definition can be kept in an internal header, with only a pointer type definition kept in a public header, in which case the structure definition can be changed however needed.

https://clusterlabs.org/projects/pacemaker/doc/3.0/Pacemaker_Development/singlehtml/index.html#structures

I wouldn't have been surprised if changing the struct definition risked breaking ABI compatibility somehow. But if Ken said it's safe to change something, then it ought to be.

Reply:
Okay, I'll buy that.